trie-db/src/proof/generate.rs

```diff
@@ -75,13 +88,34 @@ impl<'a, C: NodeCodec> StackEntry<'a, C> {
 
 	/// Encode this entry to an encoded trie node with data properly omitted.
 	fn encode_node(mut self) -> TrieResult<Vec<u8>, C::HashOut, C::Error> {
+		let omit_value = self.omit_value;
+		let prepend_on_hashed = |mut encoded: Vec<u8>| -> Vec<u8> {
+			if let Some(header) = C::ESCAPE_HEADER {
+				if matches!(omit_value, ValueMgmt::OmitHashedValue) {
+					for i in header.iter() {
+						encoded.insert(0, *i);
+					}
```

Comment:
is it intentional to write it in reversed ?

Reply:
also inserting at index 0 isn't very efficient, maybe it is better to create a new vec rather than moving all elements by one every time.

Reply:
It is adding a escape header sequence.
Compact proof got a special encoding for node that are skipped.
Value skipped are 0 length encoded, and build again when verifying (we verify with the value as parameter).
Problem is we do not know if the value is hashed or not (as long as the trie state can be hybrid), so when hashed
we prepend this sequence).
In practice this sequence is only one byte.

Reply:
I am just thinking, the fact that we will migrate state, we can maybe remove this code and apply threshold when verifying to know if we want to hash?

Reply:
> In practice this sequence is only one byte.

I now read that in verify we check the header is in order, I think we should just make ESCAPE_HEADER a `Option<u8>` to fix this

Reply:
Oh yes, you are right insert should be rev, I will switch to single byte (if needed sequence could be use in the future).

Reply:
> I am just thinking, the fact that we will migrate state, we can maybe remove this code and apply threshold when verifying to know if we want to hash?

But until it is all migrated we still need this no ?

Reply:
This proof is not use a lot, the only part of code I know that use it is fastsync or warpsynch, but some of their use case do not really work with hybrid.
I don't know of any other usage of this compact proof (the one for cumulus is trie_codec.rs version which really need escaping).